code/obj/machinery/singularity.dm

```diff
@@ -401,6 +401,8 @@ for some reason I brought it back and tried to clean it up a bit and I regret ev
 			if (T.turf_flags & IS_TYPE_SIMULATED)
 				if (istype(T,/turf/simulated/floor) && !istype(T,/turf/simulated/floor/plating))
 					var/turf/simulated/floor/F = T
+					if (istype(F, /turf/simulated/floor/glassblock))
+						return
```

Comment:
I'd rather it just remove the floor completely, rather than making it so singularities can't eat glass floors.

Reply:
This was just meant to prevent them from generating floor tiles at a distance but idk if that was needed actually